src/avx512-64bit-keyvaluesort.hpp

```diff
@@ -0,0 +1,1276 @@
+/*******************************************************************
+ * Copyright (C) 2022 Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ * Authors: Raghuveer Devulapalli <raghuveer.devulapalli@intel.com>
+ * ****************************************************************/
+
+#ifndef AVX512_QSORT_64BIT_KV
+#define AVX512_QSORT_64BIT_KV
+
+#include "avx512-common-keyvaluesort.h"
+
+/*
+ * Constants used in sorting 8 elements in a ZMM registers. Based on Bitonic
+ * sorting network (see
+ * https://en.wikipedia.org/wiki/Bitonic_sorter#/media/File:BitonicSort.svg)
+ */
+// ZMM                  7, 6, 5, 4, 3, 2, 1, 0
+#define NETWORK_64BIT_1 4, 5, 6, 7, 0, 1, 2, 3
+#define NETWORK_64BIT_2 0, 1, 2, 3, 4, 5, 6, 7
+#define NETWORK_64BIT_3 5, 4, 7, 6, 1, 0, 3, 2
+#define NETWORK_64BIT_4 3, 2, 1, 0, 7, 6, 5, 4
+
+template <>
+struct zmm_kv_vector<int64_t> {
```

Comment:
Is there a difference between `zmm_kv_vector `struct and `zmm_vector `struct in `avx512-64bit-qsort.hpp`? If not, we should move all these to a new header file `avx512-64bit-common.h` and not duplicate them. You could include the NETWORK_64BIT_* constants too.